### ISSUE TITLE
Using the gg-13-8-99-515-806-nb-classifier.qza classfier results in a warning

### ISSUE THREAD

**Issue by johnchase** [qiime2/q2-feature-classifier#42]
Using the [gg-13-8-99-515-806-nb-classifier.qza](https://docs.qiime2.org/2.0.6/data/tutorials/moving-pictures/gg-13-8-99-515-806-nb-classifier.qza) classifier  results in an warning that the classifier was created with an older version of scikit-learn than what is currently on my system. This is a small issue, though I figured it was worth noting. 

```python
taxa_assignments = feature_classifier.methods.classify(reads=qiime_seqs, classifier=taxa_classifier, n_jobs=14)

 UserWarning: Trying to unpickle estimator FeatureHasher from version pre-0.18 when using version 0.18. This might lead to breaking code or invalid results. Use at your own risk.
  UserWarning)
```

**Comment by BenKaehler** [comment]
Thanks @johnchase, it is worth noting.

It is fairly straightforward to retrain the classifier with the version of sklearn that you have installed. A tutorial will soon be available in the qiime2 docs, but of the moment you could access it here:

https://github.com/BenKaehler/docs/blob/ed22c7d8026af39825568a5f13ad904426602a8d/source/tutorials/feature-classifier.rst

You may wish to view it as raw so that you can see the commands.

In the mean time, if you notice any strange behaviour that might be attributable to the cross-version pickles, please let us know.

**Comment by BenKaehler** [comment]
We discussed it last year in the developers' meeting. From memory, at that meeting we decided that we should move away from using pickle. We could do that by taking explicit responsibility for saving down trained classifier parameters for specific classifiers, and not allowing users to save experimental classifiers by rolling classifier training and classification into a single step.

However, we will still have a dependency on the sklearn version. I don't see that we can move away from that and we should encourage users to always train there own classifiers for their specific problem.
